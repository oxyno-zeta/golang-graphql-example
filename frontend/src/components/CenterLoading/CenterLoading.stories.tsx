import React from 'react';
import { StoryFn, Meta } from '@storybook/react-vite';
import Stack from '@mui/material/Stack';
import { useTranslation } from 'react-i18next';
import CenterLoading, { Props } from './CenterLoading';

export default {
  title: 'Components/CenterLoading',
  component: CenterLoading,
} as Meta<typeof CenterLoading>;

export const Playground: StoryFn<typeof CenterLoading> = function C(args: Props) {
  return <CenterLoading {...args} />;
};

export const Colors: StoryFn<typeof CenterLoading> = function C() {
  return (
    <Stack spacing={2}>
      <CenterLoading />
      <CenterLoading circularProgressProps={{ color: 'secondary' }} />
      <CenterLoading circularProgressProps={{ color: 'success' }} />
      <CenterLoading circularProgressProps={{ color: 'error' }} />
    </Stack>
  );
};

export const Subtitles: StoryFn<typeof CenterLoading> = function C() {
  const { t } = useTranslation();
  return (
    <Stack spacing={2}>
      <CenterLoading subtitle={t('common.loadingText')} />
      <CenterLoading circularProgressProps={{ color: 'error' }} subtitle={t('common.errors')} />
    </Stack>
  );
};
