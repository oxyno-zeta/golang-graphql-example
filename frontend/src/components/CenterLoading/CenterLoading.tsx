import React from 'react';
import Box from '@mui/material/Box';
import CircularProgress, { CircularProgressProps } from '@mui/material/CircularProgress';
import type { SxProps, TypographyProps } from '@mui/material';
import Typography from '@mui/material/Typography';

export interface Props {
  /**
   * Container Box sx.
   * Following this documentation: https://mui.com/system/getting-started/the-sx-prop/ .
   */
  readonly containerBoxSx?: SxProps;
  /**
   * Circular progress props.
   * Following this documentation: https://mui.com/material-ui/api/circular-progress/#props .
   */
  readonly circularProgressProps?: Partial<CircularProgressProps>;
  /**
   * subtitle.
   */
  readonly subtitle?: string;
  /**
   * Typography props.
   * Following this documentation: https://mui.com/material-ui/api/typography/#props .
   */
  readonly subtitleTypographyProps?: Partial<TypographyProps>;
}

function CenterLoading({
  containerBoxSx = {},
  circularProgressProps = {},
  subtitle = undefined,
  subtitleTypographyProps = {},
}: Props) {
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        margin: '10px 0',
        ...containerBoxSx,
      }}
    >
      <CircularProgress {...circularProgressProps} />
      {subtitle ? (
        <Typography
          color="text.secondary"
          sx={{ fontSize: '12px', marginTop: '7px' }}
          variant="subtitle2"
          {...subtitleTypographyProps}
        >
          {subtitle}
        </Typography>
      ) : null}
    </Box>
  );
}

export default CenterLoading;
