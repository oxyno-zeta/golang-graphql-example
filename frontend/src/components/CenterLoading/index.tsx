import CenterLoading, { Props } from './CenterLoading';

export default CenterLoading;
export type { Props };
