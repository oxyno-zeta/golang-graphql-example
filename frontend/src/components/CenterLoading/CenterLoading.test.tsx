import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import CenterLoading from './CenterLoading';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('CenterLoading', () => {
  it('should display main page center loading', async () => {
    const { container, findByRole } = render(<CenterLoading />);

    // Find progressbar
    expect(await findByRole('progressbar')).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
