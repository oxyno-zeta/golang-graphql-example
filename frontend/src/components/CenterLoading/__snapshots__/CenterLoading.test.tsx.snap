// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterLoading should display main page center loading 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin: 10px 0;
}

.emotion-1 {
  display: inline-block;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
  color: #1976d2;
}

.emotion-2 {
  display: block;
}

.emotion-3 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary  emotion-1"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg  emotion-2"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate  emotion-3"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
